numba/tests/test_record_dtype.py

```diff
@@ -293,6 +294,36 @@ class Ref(ctypes.Structure):
         dtype = ty.dtype
         self.assertTrue(dtype.isalignedstruct)
 
+    def test_complex_struct(self):
+        class Complex(ctypes.Structure):
+            _fields_ = [
+                ('real', ctypes.c_double),
+                ('imag', ctypes.c_double),
+            ]
+
+        class Ref(ctypes.Structure):
+            _fields_ = [
+                ('apple', ctypes.c_int32),
+                ('mango', Complex),
+            ]
+
+        ty = types.Record.make_c_struct([
+            ('apple', types.intc),
+            ('mango', types.complex128),
+        ])
+        # Correct offsets
+        self.assertEqual(len(ty), 2)
+        self.assertEqual(ty.offset('apple'), Ref.apple.offset)
+        self.assertEqual(ty.offset('mango'), Ref.mango.offset)
+        # Correct size
+        self.assertEqual(ty.size, ctypes.sizeof(Ref))
+        # Is aligned?
+        # NumPy version < 1.16 misalign complex-128 types to 16bytes,
+        # so don't check the dtype there.
+        if numpy_version >= (1, 16):
```

Comment:
This may need some discussion

Reply:
Wondering if in the code a warning should be raised if this behaviour and np<116 is detected?

Reply:
I've made it compare alignment reported by numpy and llvm irrespective of the numpy version, and it will raise exception if a `.dtype` is requested by a `Record` and the alignments are incompatible.

Reply:
That's a good idea, probably most safe.